agent/src/agent/destination/http.py

```diff
@@ -67,81 +85,173 @@ def token(self):
-        self.config[self.CONFIG_PROXY_USE] = enable
-        if enable:
-            self.config[self.CONFIG_PROXY_URI] = uri
-            self.config[self.CONFIG_PROXY_USERNAME] = username
-            self.config[self.CONFIG_PROXY_PASSWORD] = password
+    @property
+    def proxy(self) -> Optional[Proxy]:
+        if self.config[self.CONFIG_PROXY_USE]:
+            return Proxy(
+                self.config[self.CONFIG_PROXY_URI],
+                self.config[self.CONFIG_PROXY_USERNAME],
+                self.config[self.CONFIG_PROXY_PASSWORD],
+            )
+        return None
+
+    @proxy.setter
+    def proxy(self, proxy: Proxy):
+        self.config[self.CONFIG_PROXY_USE] = True
+        self.config[self.CONFIG_PROXY_URI] = proxy.uri
+        self.config[self.CONFIG_PROXY_USERNAME] = proxy.username
+        self.config[self.CONFIG_PROXY_PASSWORD] = proxy.password
 
     def get_proxy_url(self) -> str:
-        return self.config.get(self.CONFIG_PROXY_URI)
+        return self.config.get(self.CONFIG_PROXY_URI, '')
 
     def get_proxy_username(self) -> str:
-        return self.config.get(self.CONFIG_PROXY_USERNAME)
+        return self.config.get(self.CONFIG_PROXY_USERNAME, '')
 
-    def get_proxy_password(self) -> str:
-        return self.config.get(self.CONFIG_PROXY_PASSWORD)
 
-    def get_proxy_configs(self) -> dict:
-        proxies = dict()
-        if self.config[self.CONFIG_PROXY_USE]:
-            proxy_parsed = urlparse(self.get_proxy_url())
-            netloc = proxy_parsed.netloc
-            if self.get_proxy_password():
-                netloc = self.get_proxy_username() + ':' + self.get_proxy_password() + '@' + netloc
-            proxies['http'] = urlunparse((proxy_parsed.scheme, netloc, proxy_parsed.path, '', '', ''))
-            proxies['https'] = proxies['http']
-        return proxies
-
-    def validate_token(self) -> bool:
-        result = requests.post(self.config[self.CONFIG_RESOURCE_URL], proxies=self.get_proxy_configs(), timeout=5)
-        result.raise_for_status()
-        return True
+class DestinationException(click.ClickException):
+    pass
+
 
-    def validate_api_key(self) -> bool:
-        if self.api_key:
-            AnodotApiClient(self.api_key, self.get_proxy_configs(), base_url=self.url)
+def build_urls(destination_url: str, data_collection_token: str) -> (str, str):
+    resource_url = urllib.parse.urljoin(
+        destination_url, f'api/v1/metrics?token={data_collection_token}&protocol={HttpDestination.PROTOCOL_20}')
+    monitoring_url = urllib.parse.urljoin(destination_url, f'api/v1/agents?token={data_collection_token}')
+    return resource_url, monitoring_url
+
+
+def create(
+    token: str,
+    url: str,
+    access_key: str = None,
+    proxy_host: str = None,
+    proxy_username: str = None,
+    proxy_password: str = None,
+    host_id: str = None,
+) -> Result[HttpDestination, str]:
+    return __build(HttpDestination.get_or_default(), token, url, access_key,
```

Comment:
Why when we create a destination you use `get_or_default`? Shouldn't it just create a new empty object?

Reply:
hmm, yes, it should